Tools/scripts/run_tests.py

```diff
@@ -56,20 +53,13 @@ def main(regrtest_args):
     args.extend(test.support.args_from_interpreter_flags())
 
     args.extend(['-m', 'test',    # Run the test suite
-                 '-r',            # Randomize test order
-                 '-w',            # Re-run failed tests in verbose mode
+                 '--fast-ci',     # Fast Continuous Integration mode
```

Comment:
is this supposed to be here?  I see it called from the `Makefile` with `--slow-ci` and `--fast-ci` -- but wouldn't this override either of those?

Reply:
``make test`` runs ``python -E -m test --fast-ci --fast-ci --timeout=``. Passing the option twice is fine.

``make buildbottest`` runs ``python -E -m test --fast-ci --slow-ci --timeout=``. If regrtest gets ``--fast-ci`` **and** ``--slow-ci``, ``--slow-ci`` has the priority.

I would prefer to not duplicate these fast/slow CI options. I would also prefer to remove the run_tests.py script, but it's needed for cross-compiled Python.

It doesn't look too complicated to move run_tests.py code into libregrtest.

Reply:
makes sense -- just double checking since I got a weird conflict here (debian/deadsnakes patches out the randomization because it makes the builds nondeterministic)

Reply:
I created issue gh-110152 for that.

Reply:
> makes sense -- just double checking since I got a weird conflict here (debian/deadsnakes patches out the randomization because it makes the builds nondeterministic)

Do you set SOURCE_DATE_EPOCH env var in this case? https://reproducible-builds.org/docs/source-date-epoch/

Maybe regrtest can detect SOURCE_DATE_EPOCH and omit --randomize in this case?

Reply:
If SOURCE_DATE_EPOCH env var is set, open an issue, and I will update regrtest for that ;-)

Reply:
yep!  looks like launchpad sets that at the very least -- I assume the debian build system would too

Reply:
@asottile created https://github.com/python/cpython/issues/110164 follow-up issue.

Reply:
> yep! looks like launchpad sets that at the very least -- I assume the debian build system would too

Yes, it's set by dpkg-buildpackage in all package builds.